.github/workflows/dapr.yml

```diff
@@ -173,3 +190,71 @@ jobs:
             --body "${RELEASE_BODY}" \
             --prerelease true \
             ${RELEASE_ARTIFACT[*]}
+      - name: Package Helm chart
+        env:
+              HELM_CHARTS_DIR: charts/dapr
+        run: |
+          if [ "$LATEST_RELEASE" = "true" ]; then
+            mkdir -p ${{ env.ARTIFACT_DIR }}/${{ env.HELM_PACKAGE_DIR }}
+            helm package ${{ env.HELM_CHARTS_DIR }} --destination ${{ env.ARTIFACT_DIR }}/${{ env.HELM_PACKAGE_DIR }}
+          fi
+      - name: Upload Helm charts package to artifacts
+        uses: actions/upload-artifact@master
+        with:
+          name: dapr_helm_charts_package
+          path: ${{ env.ARTIFACT_DIR }}/${{ env.HELM_PACKAGE_DIR }}
+  helmpublish:
+    name: Publish helm charts to Helm github pages repo
+    needs: publish
+    if: github.event_name != 'pull_request'
+    env:
+      ARTIFACT_DIR: ./release
+      DAPR_VERSION_ARTIFACT: dapr_version
+      DAPR_RELEASE_TAG_ARTIFACT: dapr_release
+      HELM_PACKAGE_DIR: helm
+    runs-on: ubuntu-latest
+    steps:
+      - name: Create Helm charts directory
+        run: |
+          mkdir -p ${{ env.ARTIFACT_DIR }}/${{ env.HELM_PACKAGE_DIR }}
+      - name: download artifacts - dapr_helm_charts_package
+        uses: actions/download-artifact@master
+        with:
+          name: dapr_helm_charts_package
+          path: ${{ env.ARTIFACT_DIR }}/${{ env.HELM_PACKAGE_DIR }}
+      - name: download artifacts - dapr_version
+        uses: actions/download-artifact@master
+        with:
+          name: ${{ env.DAPR_VERSION_ARTIFACT }}
+          path: ${{ env.ARTIFACT_DIR }}/${{ env.DAPR_VERSION_ARTIFACT }}
+      - name: download artifacts - dapr_release
+        uses: actions/download-artifact@master
+        with:
+          name: ${{ env.DAPR_RELEASE_TAG_ARTIFACT }}
+          path: ${{ env.ARTIFACT_DIR }}/${{ env.DAPR_RELEASE_TAG_ARTIFACT }}
+      - name: Checkout Helm Charts Repo
+        uses: actions/checkout@v2
+        env:
+              DAPR_HELM_REPO: dapr/dapr-helm-charts.io
+              DAPR_HELM_REPO_CODE_PATH: dapr-helm-charts
+        with:
+          repository: ${{ env.DAPR_HELM_REPO }}
+          ref: refs/heads/master
+          token: ${{ secrets.DAPR_BOT_TOKEN }}
+          path: ${{ env.DAPR_HELM_REPO_CODE_PATH }}
+      - name: Upload helm charts to Helm Repo
+        env:
+              DAPR_HELM_REPO_CODE_PATH: dapr-helm-charts
+              DAPR_VERSION_ARTIFACT: dapr_version
+        run: |
+            daprReleaseTag=`cat ${{ env.ARTIFACT_DIR }}/${{ env.DAPR_RELEASE_TAG_ARTIFACT }}/${{ env.DAPR_RELEASE_TAG_ARTIFACT }}`
+            if [ "$daprReleaseTag" = "true" ]; then
```

Comment:
this check can be moved earlier, if its not true, then checkout of helm charts repo is not needed.

Reply:
@vinayada1 please see my comment above. we can remove this if.

Reply:
fixed and removed dapr_release artifact altogether since it was used only for this if check